.github/workflows/build-images-pr.yaml

```diff
@@ -0,0 +1,15 @@
+name: PR Conditional Build and Push Image to Quay.io Repo
+
+on:
+  pull_request_target:
+    types: [labeled, opened, synchronize, reopened]
+
+jobs:
+  workflow-build:
+    if: ${{ github.event_name == 'pull_request_target' && contains(github.event.pull_request.labels.*.name, 'forked_image_approved') }}
+    name: Calls build-images-base workflow
+    uses: ./.github/workflows/build-images-base.yaml
+    secrets: inherit
+    with:
+      kuadrantOperatorVersion: ${{ github.event.pull_request.user.login }}-${{ github.event.pull_request.number }}
+      kuadrantOperatorTag: ${{ github.event.pull_request.user.login }}-${{ github.event.pull_request.number }}
```

Comment:
I think we should also consider to always remove the `forked_image_approved` label regardless of whether the image build was successfull or not as an additional security measure - https://github.com/actions-ecosystem/action-remove-labels.

And also possibly disallow multiples of this workflow from running concurrently (i.e. I'm thinking someone might first get the label and then quickly push something that could leak our secrets so we should try to prevent/mitigate this from happening) 🤔  - https://docs.github.com/en/actions/using-jobs/using-concurrency